### DISCUSSION TITLE
How to actually disable a builtin extension.

### DISCUSSION THREAD

**Discussion by MrGingerSir222** [vladmandic/sdnext#1284] [Q&A]
I keep going to the extensions and trying to disable sd-webui-agent-scheduler and multidiffusion-upscaler-for-automatic1111, but every time I start up the program again, it re-enables them. And if I uninstall something else, or make any changes, it also re-enables them. 
Is there a way to truly disable them?

**Comment by brknsoul** [comment]
I've noticed this too. It makes my UI feel cluttered with builtins i never use.

**Comment by vladmandic** [comment]
if you go to extensions, unselect one you want to disable and click apply, it should work. 
i just tried with controlnet and `config.json` was correctly updated:

```json
    "disabled_extensions": [
        "sd-webui-controlnet",
    ],
```

note that extension may still appear in ui unless you do a manual server restart as its near-impossible to get python to unload some stuff from memory (adding new stuff is easy so you can install extension and use restart from ui, but disable is tricky).

once extension is disabled, its never auto-enabled by app.
and even if all extensions are updated, it still stays disabled.

**Reply by MrGingerSir222** [reply]
This is what I was doing.
lol, I swear it wasn't working before, but now it is. 
haha thanks

**Reply by Woisek** [reply]
> once extension is disabled, its never auto-enabled by app. and even if all extensions are updated, it still stays disabled.

I experienced the opposite right now.
I disabled the agent scheduler and after the last restart, it was enabled again. 🤨

**Reply by MrGingerSir222** [reply]
> > once extension is disabled, its never auto-enabled by app. and even if all extensions are updated, it still stays disabled.
> 
> I experienced the opposite right now. I disabled the agent scheduler and after the last restart, it was enabled again. 🤨

Yeah, I’m actually still having this problem occasionally too yet.

**Reply by Woisek** [reply]
Yes, and today again. 😕 That's a bit annoying ...

**Reply by ghost** [reply]
It does not return for me, what I did was deleting the builtin-extension from the folder, and then deactivating it in the ui. I really like that extension but I had the ui freeze bug (model selection was nonstop loading too) and Im going to wait until that is resolved and then I will enable it again

**Reply by Woisek** [reply]
I also deleted it, but after the next update it was again installed and activated again ... 🤨

**Reply by vladmandic** [reply]
Well, don't delete, just disable.

**Reply by Woisek** [reply]
I also did that, but it keeps enabling itself.

**Reply by Woisek** [reply]
Did an update and .. the f* agent scheduler is active again ...
I can now definitely say, this statement:

>once extension is disabled, its never auto-enabled by app.

is not true, sorry. 😔

**Reply by ghost** [reply]
I read it now has a task save option which is something I always wanted implemented. I think I will try to enable it again. I have no problem with it and I want to use it as long as it does not freeze my ui. Going to try that right now

edit: ok doesnt work

**Comment by TeutonJon78** [comment]
It normally works from the GUI extensions page, but it seems the agent-scheduler is locking up the UI for many people. If you just disable that extensions and quit and reload the UI, you can disable extensions like normal from the UI again.

Edit: it also seems like it might be related to using a non-"black and orange" theme. And some people seem to be able to fix it by reloading the browser after a bit.

**Reply by vladmandic** [reply]
that is specific to usage of `--autolaunch` since it starts browser too early so browser goes unresponsive due to too many DOM changes still happening. i've changed code yesterday so browser launch starts later. in either case, not really related to can you disable extension.
